### PR TITLE
[WebPubSub] Support new release 2022-11-01

### PR DESCRIPTION
**Pull request by xingsy97**
# Description

Support [new release 2022-11-01](https://github.com/Azure/azure-rest-api-specs/pull/20762)

1. Support sending with "filter" syntax
2. Support generating access url with initial groups
3. Add new API "removeConnectionFromAllGroups"
4. Add patches for `send_to_all`, `send_to_user`, `send_to_connection`, `send_to_group` and their async versions. Because the new version of autorest python requires sdk manager to pre-process http request content according to different `Content-Type` by themselves.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by msyyc** [comment]
@xingsy97 Which version of cmd do you use to update the SDK code?

**Comment by xingsy97** [comment]
> @xingsy97 Which version of cmd do you use to update the SDK code?

Hi @msyyc , I use `autorest --version=3.9.2 --use=@autorest/python@6.1.11 --use=@autorest/modelerfour@4.24.3 ./swagger/README.md`

File: sdk/webpubsub/azure-messaging-webpubsubservice/samples/integration_sample.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+time.sleep(5)
+
+print("Received Message =", recv_messages)
+assert service.group_exists("InitGroup") == True
+assert recv_messages.count("Message_For_All") == WEBSOCKET_N
+assert recv_messages.count("Message_For_InitGroup") == WEBSOCKET_N
+assert recv_messages.count("Message_Not_For_User0") == WEBSOCKET_N - 1
+assert recv_messages.count("Message_For_User0") == 1
+assert recv_messages.count("Message_For_RemoveFromAll") == 3
```
Comment (msyyc) [inline sdk/webpubsub/azure-messaging-webpubsubservice/samples/integration_sample.py:85]: The sample is complicated so I advise to use standard structure including `main()`like: https://github.com/Azure/azure-sdk-for-python/blob/38271d04d5f8175ab2a32612eb15d930203f8afe/sdk/webpubsub/azure-mgmt-webpubsub/generated_samples/operations_list.py#L26-L39

Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/samples/integration_sample.py]: Updated this sample

**Review by msyyc — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Comment by msyyc** [comment]
Since @iscai-msft is on vacation, add @lmazuel to review

File: sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Release History
 
+## 1.1.0b1 (2022-11-01)
+
+### Features Added
+
+- Support sending with "filter" syntax
```
Comment (iscai-msft) [inline sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md:7]: the changelog should focus more on the changes that python sdk writers will encounter, so it should be about which parameters / functions are added in python syntax

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py [hunk 3]
```diff
@@ -55,14 +57,71 @@
...
+
     @distributed_trace_async
     async def get_client_access_token(
         self,
         *,
         user_id: Optional[str] = None,
         roles: Optional[List[str]] = None,
-        minutes_to_expire: Optional[int] = 60,
+        minutes_to_expire: int = 60,
+        group: Optional[List[str]] = None,
```
Comment (iscai-msft) [inline sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py:124]: mention the added parameter in the changelog as well

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py [hunk 4]
```diff
@@ -106,14 +109,14 @@ def build_close_all_connections_request(
 
 
 def build_send_to_all_request(
-    hub: str, *, json: Any = None, content: Any = None, excluded: Optional[List[str]] = None, **kwargs: Any
+    hub: str, *, content: IO, excluded: Optional[List[str]] = None, filter_string: Optional[str] = None, **kwargs: Any
```
Comment (iscai-msft) [inline sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py:112]: should this be called `filter` instead?

**Review by iscai-msft — COMMENTED** [review]
i think overall it looks good, hard to tel though bc all of the functions switched order

File: sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md (hunk 2 above)
Comment (msyyc) [reply sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md:7]: You can follow the pattern like: `operation xxx has new optional parameter filter`

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py (hunk 3 above)
Comment (msyyc) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py:124]: here is some pattern to refer: https://github.com/Azure/azure-sdk-for-python/blob/75b9cb73b2c9edfcecaa05c6e03e52a7ea805a30/tools/azure-sdk-tools/packaging_tools/change_log.py#L198-L215

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py (hunk 4 above)
Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py:112]: `filter` cannot pass CI check, due to name conflict with python built-in function `filter`. So I renamed it to `filter_string`.

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py (hunk 3 above)
Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/aio/_operations/_operations.py:124]: Updated

File: sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md (hunk 2 above)
Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md]: Updated

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py (hunk 4 above)
Comment (msyyc) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py:112]: @xingsy97 Could you show the CI failure about `filter`? The name `filter` is widely used in many packages and we could suppress the check if necessary

Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_operations.py]: run pylint 
`python G:\azure-sdk-for-python\eng\tox\run_pylint.py -t G:\azure-sdk-for-python\sdk\webpubsub\azure-messaging-webpubsubservice`

See errors like
```
azure\messaging\webpubsubservice\aio\_operations\_patch.py:336: [W0622(redefined-builtin), WebPubSubServiceClientOperationsMixin.send_to_user] Redefining built-in 'filter'
azure\messaging\webpubsubservice\_operations\_operations.py:773: [W0622(redefined-builtin), WebPubSubSerubServiceClientOperationsMixin.send_to_all] Redefining built-in 'filter'
```

File: sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,14 @@
 # Release History
 
+## 1.1.0b1 (2022-11-01)
+
+### Features Added
+- Operation `send_to_all` has a new optional parameter `filter_string`
+- Operation `send_to_user` has a new optional parameter `filter_string`
+- Operation `send_to_group` has a new optional parameter `filter_string`
+- Operation `get_client_access_token` has a new optional parameter `group`
+- Added operation `removeConnectionFromAllGroups`
```
[the remark this answers is not in the document]
Comment (xingsy97) [reply sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md]: Updated

File: tools/azure-sdk-tools/ci_tools/environment_exclusions.py [hunk 6]
```diff
@@ -61,6 +61,7 @@
     "azure-agrifood-farming",
     "azure-developer-loadtesting",
     "azure-developer-devcenter",
+    "azure-messaging-webpubsubservice",
```
Comment (msyyc) [inline tools/azure-sdk-tools/ci_tools/environment_exclusions.py:64]: will remove it after codegen fix the problem

**Comment by msyyc** [comment]
CI failure is not caused by this PR and it can be ignored.

**Pull request merged**